### PR TITLE
Add enable security nesting input to itest

### PR DESCRIPTION
**Pull request by kelkawi-a**
### Overview

Addresses #441 by adding an input parameter to the integration tests for enabling security nesting when building rocks.

### Rationale

This is needed for rocks with nested containers (e.g. docker containers) as part of the build process.

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: internal/build/action.yaml [hunk 1]
```diff
@@ -26,6 +26,8 @@ inputs:
   rockcraft-repository:
     description: Pull and build rockcraft from source instead of using snapstore version (this means that the rockcraft-channel input will be ignored).
     default: ""
+  rockcraft-enable-security-nesting:
+    description: Set security.nesting=true on the rockcraft lxc project.
```
Comment (yanksyoon) [inline internal/build/action.yaml:30]: Hello! Would you be able help briefly elaborate what security.nesting=True on lxc project does please?

File: src/build.ts
Comment (yanksyoon) [inline src/build.ts:1]: I think you may want to build and push the build dists here as well, would you be able to run `npm run bundle` to do so please?

Comment (alithethird) [reply src/build.ts:1]: I agree you should build and add it into the PR, I was not able to find a CI run that builds dists here.

File: internal/build/action.yaml (hunk 1 above)
Comment (kelkawi-a) [reply internal/build/action.yaml]: I've updated the description, [security nesting](https://discuss.linuxcontainers.org/t/what-does-security-nesting-true/7156/2) can be used to allow containers to spin up other containers. This is needed when building the [Airbyte rock](https://github.com/canonical/airbyte-k8s-operator) as it spins up a docker image inside the rock as part of the build process.

File: src/build.ts
Comment (kelkawi-a) [reply src/build.ts:1]: Done :) though i'm not sure if I've handled the boolean casting correctly in the TS code, could you please let me know?

Comment (alithethird) [reply src/build.ts:1]: This is a good concern. We need to make sure it's done correctly.

Comment (alithethird) [reply src/build.ts:1]: I think you should use `getBooleanInput` according to https://github.com/actions/toolkit/tree/main/packages/core#inputsoutputs

**Review by alithethird — COMMENTED** [review]
Please don't merge yet.

**Pull request closed without merging**